File: src/Application Specific/Classes/Entity Collections/Order/OrderCharge/OrderChargeSQL.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace OrderCharges

Public Class OrderChargeSQL
    Private _database As Sys.Database

    Public Sub New(ByVal database As Sys.Database)
        _database = database
    End Sub

#Region "Functions"

	
    Public Sub Delete(ByVal OrderChargeID As Integer)
    		_database.ClearParameter()
            _database.AddParameter("@OrderChargeID", OrderChargeID, True)
            _database.ExecuteSP_NO_Return("OrderCharge_Delete")
    End Sub

            Public Function OrderCharge_GetForOrder(ByVal OrderID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@OrderID", OrderID, True)
                Dim dt As DataTable = _database.ExecuteSP_DataTable("OrderCharge_GetForOrder")
                dt.TableName = Entity.Sys.Enums.TableNames.tblOrderCharge.ToString
                Return New DataView(dt)
            End Function

            Public Function [OrderCharge_Get](ByVal OrderChargeID As Integer) As OrderCharge
                _database.ClearParameter()
                _database.AddParameter("@OrderChargeID", OrderChargeID)

                'Get the datatable from the database store in dt
                Dim dt As DataTable = _database.ExecuteSP_DataTable("OrderCharge_Get")

                If Not dt Is Nothing Then
                    If dt.Rows.Count > 0 Then
                        Dim oOrderCharge As New OrderCharge
                        With oOrderCharge
                            .IgnoreExceptionsOnSetMethods = True
                            .SetOrderChargeID = dt.Rows(0).Item("OrderChargeID")
                            .SetOrderID = dt.Rows(0).Item("OrderID")
                            .SetOrderChargeTypeID = dt.Rows(0).Item("OrderChargeTypeID")
                            .SetAmount = dt.Rows(0).Item("Amount")
                            .SetReference = dt.Rows(0).Item("Reference")

                            .SetDeleted = dt.Rows(0).Item("Deleted")
                        End With
                        oOrderCharge.Exists = True
                        Return oOrderCharge
                    Else
                        Return Nothing
                    End If
                Else
                    Return Nothing
                End If
            End Function

            Public Function OrderCharge_GetForConsolidatedOrder(ByVal OrderConsolidationID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@OrderConsolidationID", OrderConsolidationID, True)
                Dim dt As DataTable = _database.ExecuteSP_DataTable("OrderCharge_GetForConsolidatedOrder")
                dt.TableName = Entity.Sys.Enums.TableNames.tblOrderCharge.ToString
                Return New DataView(dt)
            End Function



            Public Function [OrderCharge_GetAll]() As DataView
                _database.ClearParameter()
                Dim dt As DataTable = _database.ExecuteSP_DataTable("OrderCharge_GetAll")
                dt.TableName = Entity.Sys.Enums.TableNames.tblOrderCharge.ToString
                Return New DataView(dt)
            End Function

            Public Function [Insert](ByVal oOrderCharge As OrderCharge) As OrderCharge
                _database.ClearParameter()
                AddOrderChargeParametersToCommand(oOrderCharge)

                oOrderCharge.SetOrderChargeID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("OrderCharge_Insert"))
                oOrderCharge.Exists = True
                Return oOrderCharge
            End Function


            Public Sub [Update](ByVal oOrderCharge As OrderCharge)
                _database.ClearParameter()
                _database.AddParameter("@OrderChargeID", oOrderCharge.OrderChargeID, True)
                AddOrderChargeParametersToCommand(oOrderCharge)
                _database.ExecuteSP_NO_Return("OrderCharge_Update")
            End Sub



            Private Sub AddOrderChargeParametersToCommand(ByRef oOrderCharge As OrderCharge)
                With _database
                    .AddParameter("@OrderID", oOrderCharge.OrderID, True)
                    .AddParameter("@OrderChargeTypeID", oOrderCharge.OrderChargeTypeID, True)
                    .AddParameter("@Amount", oOrderCharge.Amount, True)
                    .AddParameter("@Reference", oOrderCharge.Reference, True)



                End With
            End Sub


#End Region

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Order/OrderCharge/OrderChargeSQL.cs
using System.Data;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace OrderCharges
    {
        public class OrderChargeSQL
        {
            private Sys.Database _database;

            public OrderChargeSQL(Sys.Database database)
            {
                _database = database;
            }

            

            public void Delete(int OrderChargeID)
            {
                _database.ClearParameter();
                _database.AddParameter("@OrderChargeID", OrderChargeID, true);
                _database.ExecuteSP_NO_Return("OrderCharge_Delete");
            }

            public DataView OrderCharge_GetForOrder(int OrderID)
            {
                _database.ClearParameter();
                _database.AddParameter("@OrderID", OrderID, true);
                var dt = _database.ExecuteSP_DataTable("OrderCharge_GetForOrder");
                dt.TableName = Sys.Enums.TableNames.tblOrderCharge.ToString();
                return new DataView(dt);
            }

            public OrderCharge OrderCharge_Get(int OrderChargeID)
            {
                _database.ClearParameter();
                _database.AddParameter("@OrderChargeID", OrderChargeID);

                // Get the datatable from the database store in dt
                var dt = _database.ExecuteSP_DataTable("OrderCharge_Get");
                if (dt is object)
                {
                    if (dt.Rows.Count > 0)
                    {
                        var oOrderCharge = new OrderCharge();
                        oOrderCharge.IgnoreExceptionsOnSetMethods = true;
                        oOrderCharge.SetOrderChargeID = dt.Rows[0]["OrderChargeID"];
                        oOrderCharge.SetOrderID = dt.Rows[0]["OrderID"];
                        oOrderCharge.SetOrderChargeTypeID = dt.Rows[0]["OrderChargeTypeID"];
                        oOrderCharge.SetAmount = dt.Rows[0]["Amount"];
                        oOrderCharge.SetReference = dt.Rows[0]["Reference"];
                        oOrderCharge.SetDeleted = Conversions.ToBoolean(dt.Rows[0]["Deleted"]);
                        oOrderCharge.Exists = true;
                        return oOrderCharge;
                    }
                    else
                    {
                        return null;
                    }
                }
                else
                {
                    return null;
                }
            }

            public DataView OrderCharge_GetForConsolidatedOrder(int OrderConsolidationID)
            {
                _database.ClearParameter();
                _database.AddParameter("@OrderConsolidationID", OrderConsolidationID, true);
                var dt = _database.ExecuteSP_DataTable("OrderCharge_GetForConsolidatedOrder");
                dt.TableName = Sys.Enums.TableNames.tblOrderCharge.ToString();
                return new DataView(dt);
            }

            public DataView OrderCharge_GetAll()
            {
                _database.ClearParameter();
                var dt = _database.ExecuteSP_DataTable("OrderCharge_GetAll");
                dt.TableName = Sys.Enums.TableNames.tblOrderCharge.ToString();
                return new DataView(dt);
            }

            public OrderCharge Insert(OrderCharge oOrderCharge)
            {
                _database.ClearParameter();
                AddOrderChargeParametersToCommand(ref oOrderCharge);
                oOrderCharge.SetOrderChargeID = Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("OrderCharge_Insert"));
                oOrderCharge.Exists = true;
                return oOrderCharge;
            }

            public void Update(OrderCharge oOrderCharge)
            {
                _database.ClearParameter();
                _database.AddParameter("@OrderChargeID", oOrderCharge.OrderChargeID, true);
                AddOrderChargeParametersToCommand(ref oOrderCharge);
                _database.ExecuteSP_NO_Return("OrderCharge_Update");
            }

            private void AddOrderChargeParametersToCommand(ref OrderCharge oOrderCharge)
            {
                {
                    var withBlock = _database;
                    withBlock.AddParameter("@OrderID", oOrderCharge.OrderID, true);
                    withBlock.AddParameter("@OrderChargeTypeID", oOrderCharge.OrderChargeTypeID, true);
                    withBlock.AddParameter("@Amount", oOrderCharge.Amount, true);
                    withBlock.AddParameter("@Reference", oOrderCharge.Reference, true);
                }
            }


            
        }
    }
}